lib/components/form/user-settings.js

```diff
@@ -1,308 +1,424 @@
+import * as userActions from '../../actions/user'
+import { getFormattedPlaces } from '../../util/i18n'
+import {
+  getPlaceDetail,
+  getPlaceMainText,
+  isHome,
+  isOtpMiddleware,
+  isWork
+} from '../../util/user'
+import { LinkWithQuery } from '../form/connected-links'
+import { StyledMainPanelPlace } from '../user/places/styled'
+import PlaceShortcut from '../user/places/place-shortcut'
 
-import { forgetSearch, toggleTracking } from '../../actions/api'
-import { setQueryParam } from '../../actions/form'
-import { forgetPlace, forgetStop, setLocation } from '../../actions/map'
-import { setViewedStop } from '../../actions/ui'
-import Icon from '../util/icon'
+import { UnpaddedList } from './styled'
 
-const { formatStoredPlaceName, getDetailText, matchLatLon } = coreUtils.map
-const { summarizeQuery } = coreUtils.query
+const { matchLatLon } = coreUtils.map
+const { hasTransit, toSentenceCase } = coreUtils.itinerary
+/**
+ * Version of summarizeQuery and helper function that supports i18n.
+ * FIXME: replace when the original does support i18n.
+ */
+function findLocationType(
+  intl,
+  location,
+  locations = [],
+  types = ['home', 'work', 'suggested']
+) {
+  const match = locations.find((l) => matchLatLon(l, location))
+  return match && types.indexOf(match.type) !== -1
+    ? getFormattedPlaces(match.type, intl)
+    : null
+}
+function summarizeQuery(query, intl, locations = []) {
+  const from =
+    findLocationType(intl, query.from, locations) ||
+    query.from.name.split(',')[0]
+  const to =
+    findLocationType(intl, query.to, locations) || query.to.name.split(',')[0]
+  const mode = hasTransit(query.mode)
+    ? intl.formatMessage({ id: 'common.modes.transit' })
+    : toSentenceCase(query.mode)
+  return intl.formatMessage(
+    { id: 'components.UserSettings.recentSearchSummary' },
+    { from, mode, to }
+  )
+}
 
-const BUTTON_WIDTH = 40
+/**
+ * Formats elapsed time from the specified timestamp to now,
+ * as in "5 hours ago" or the localized equivalent.
+ */
+function formatElapsedTime(timestamp, intl) {
+  // This text will be shown in a tooltip, therefore
+  // it is obtained using formaMessage rather than <FormattedfMessage>.
+  const fromNowDur = moment.duration(moment().diff(moment(timestamp)))
+  return intl
+    .formatMessage(
+      { id: 'common.time.fromNowUpdate' },
+      {
+        days: fromNowDur.days(),
+        hours: fromNowDur.hours(),
+        minutes: fromNowDur.minutes()
+      }
+    )
+    .trim()
+}
 
 class UserSettings extends Component {
   _disableTracking = () => {
-    const { intl, toggleTracking, user } = this.props
-    if (!user.trackRecent) return
-    const hasRecents = user.recentPlaces.length > 0 || user.recentSearches.length > 0
+    const { intl, localUser, toggleTracking } = this.props
+    const { recentPlaces, recentSearches, storeTripHistory } = localUser
+    if (!storeTripHistory) return
+    const hasRecents =
+      (recentPlaces && recentPlaces.length > 0) ||
```

Comment:
Looks like we can use optional chaining here!

<img width="106" alt="Screen Shot 2022-01-28 at 8 03 11 AM" src="https://user-images.githubusercontent.com/86619099/151551637-416f9c4f-310b-48b4-a245-f7ced106d07f.png">

Reply:
Changed in https://github.com/opentripplanner/otp-react-redux/pull/329/commits/5f9795a0b3f3911ee4173702d7a4c11a97bcb022.